optuna/integration/catboost.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any
+
+from packaging import version
+
+import optuna
+from optuna._imports import try_import
+
+
+with try_import() as _imports:
+    import catboost as cb
+
+    if version.parse(cb.__version__) < version.parse("0.26"):
+        raise ImportError(f"You don't have CatBoost installed! CatBoost version: {cb.__version__}")
+
+
+class CatBoostPruningCallback(object):
+    """Callback for catboost to prune unpromising trials.
+
+    If TrialPruned is raised at after_iteration in CatBoostPruningCallback, then catboost exits
+    with error. So we add `checked_pruned` function in which TrialPruned is raised if pruning is
+    nessasary.
+    You must call `check_pruned` after training manually unlike other pruning callbacks.
+
+    Args:
+        trial:
+            A :class:`~optuna.trial.Trial` corresponding to the current evaluation of the
+            objective function.
+        metric:
+            An evaluation metric for pruning, e.g., ``Logloss`` and ``AUC``.
+            Please refer to
+            `CatBoost reference
+            <https://catboost.ai/docs/references/eval-metric__supported-metrics.html>`_
+            for further details.
+        valid_name:
```

Comment:
If CatBoost always uses `validation` as a prefix, this part might be simplified. For example, it takes only `int`, whose default value is `0` and renames the argument name like `dataset_index` or something else from `valid_name`.

What do you think?